File: 프로그래머스/lv3/지준혁/등산코스 정하기.java
// 해결 X
class Solution {
    
    List<int[]>[] adj = new ArrayList[50004];
    public int[] solution(int n, int[][] paths, int[] gates, int[] summits) {
        
        for (int i = 0; i <= n; ++i) adj[i] = new ArrayList<>();
        
        for (int i = 0; i < paths.length; ++i) {
            adj[paths[i][0]].add(new int[]{paths[i][2], paths[i][1]});
            adj[paths[i][1]].add(new int[]{paths[i][2], paths[i][0]});
        }
        
        int[] answer = {};
        return answer;
    }
    int [] dijsktra(int st, int n, int[] gates, int[] summits) {
        int[] d = new int[n + 1];
        Arrays.fill(d, Integer.MAX_VALUE);
        PriorityQueue<int[]> pq = new PriorityQueue<>((a, b) -> Integer.compare(a[0], b[0]));
        d[st] = 0;
        
        while (!pq.isEmpty()) {
            int[] cur = pq.poll();
            int curWeight = cur[0];
            int curNode = cur[1];
            
            if (d[curNode] != curWeight) continue;
            
            for (var nxt : adj[curNode]) {
                int nxtWeight = nxt[0];
                int nxtNode = nxt[1];
                int intensity = Math.max(d[curNode], nxt)
            }
        }
    }
}

File: 프로그래머스/lv3/오하늘/등산코스 정하기.py
def solution(n, paths, gates, summits):
    answer = []
    # 조건 1. 처음시작과 끝이 같아야 함
    # 조건 2. 산봉우리는 한번만 (visited)
    # 조건 3. 출입구는 하나만
    # 최소가 중복인 경우, 번호 낮은 순
    
    # 전략 : 다익스트라 활용 gates 별로 해보고 최소 구하기
    gate = [] # gate 후보

    graph = [[] for _ in range(n+1)] # 노드
    
    visited = [False] * (n+1)
    distance = [int(1e9)] * (n+1)
    
    for _ in range(len(paths)): #정보 입력
        u,v,w = map(int, input().split())
        graph[u].append((v,w))
    
    
    # answer : [최소 산봉우리 번호와 intensity 최솟값]
    
    
    return answer